Detect libsndio; it does not ship a pkg-config file. To be sent upstream.

Index: spa/meson.build
--- spa/meson.build.orig
+++ spa/meson.build
@@ -57,6 +57,12 @@ if get_option('spa-plugins').allowed()
     cdata.set('HAVE_ALSA_UMP', true)
   endif
 
+  # sndio does not ship a pkg-config file
+  sndio_dep = cc.find_library('sndio',
+    has_headers: ['sndio.h'],
+    required: get_option('sndio'))
+  summary({'sndio': sndio_dep.found()}, bool_yn: true, section: 'Backend')
+
   bluez_dep = dependency('bluez', version : '>= 4.101', required: get_option('bluez5'))
   bluez_gio_dep = dependency('gio-2.0', required : get_option('bluez5'))
   bluez_gio_unix_dep = dependency('gio-unix-2.0', required : get_option('bluez5'))
